### PR TITLE
Adding delays to trait functions the sequencer implements

### PR DESCRIPTION
**Pull request by lukeiannucci**
Closes https://github.com/EspressoSystems/HotShot/issues/3456

### This PR: 

Add support to add delays to functions that the sequencer implements, method calls currently return immediately as of now. We want to mock behavior where one of these trait functions may take more time to complete

### CODE REVIEW

File: crates/example-types/src/block_types.rs [hunk 1]
```diff
@@ -365,3 +376,24 @@ impl Committable for TestBlockHeader {
...
+                    let sleep_in_millis = rand::thread_rng().gen_range(
+                        settings.min_time_in_milliseconds..=settings.max_time_in_milliseconds,
+                    );
+
+                    async_sleep(Duration::from_millis(sleep_in_millis)).await;
+                }
+            }
+        }
+    }
+}
```
Comment (ss-es) [inline crates/example-types/src/block_types.rs:399]: It looks like most of this code is duplicated for the various types. If it's the same thing, I think `handle_async_delay` could have a default implementation in the trait, and you could have each type just implement e.g. `settings(DelayConfig) -> Option<DelaySettings>`

Comment (lukeiannucci) [reply crates/example-types/src/block_types.rs:399]: yes it is duplicated, i thought the same thing, but left it like this incase in the future we have more specific tests

Comment (lukeiannucci) [reply crates/example-types/src/block_types.rs]: thanks for pointing this out! i realized the code was duplicated but this default method will help clean it up!

File: crates/example-types/src/testable_delay.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            2 => Some(SupportedTraitTypesForAsyncDelay::BlockHeader),
+            _ => {
+                assert_eq!(self.index, 3, "Need to ensure that newly added or removed `SupportedTraitTypesForAsyncDelay` enum is handled in iterator");
+                return None;
+            }
+        };
+        self.index += 1;
+        supported_type
+    }
+}
```
Comment (bfish713) [inline crates/example-types/src/testable_delay.rs:132]: What is the point of implementing iterator?

Comment (lukeiannucci) [reply crates/example-types/src/testable_delay.rs:132]: add the same settings for each type

Comment (lukeiannucci) [reply crates/example-types/src/testable_delay.rs:132]: basically so you dont have to do:
config.add_setting(SupportedTraitTypesForAsyncDelay::Storage, &delay_settings);
config.add_setting(SupportedTraitTypesForAsyncDelay::BlockHeader, &delay_settings);
...

**Review by bfish713 — APPROVED** [review]

**Pull request merged**